app/server/methods.coffee

```diff
@@ -10,10 +10,12 @@ Meteor.methods
 
   createForm: (surveyId, props)->
     #TODO Authenticate
+    currentFormCount = getSurveys().findOne(surveyId).forms.length or 0
     formId = getForms().insert
       name: props.name
       createdBy: @userId
       questions: []
+      order: ++currentFormCount
```

Comment:
It seems like we could avoid updating the order values of all the forms when one is deleted by setting the order of new forms to 1 + the order value of the last form.